### PR TITLE
SDIT-2244 Make the casenote mapping UUID index non-unique

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ FROM eclipse-temurin:21-jre-jammy
 LABEL maintainer="HMPPS Digital Studio <info@digital.justice.gov.uk>"
 
 ARG BUILD_NUMBER
-ENV BUILD_NUMBER ${BUILD_NUMBER:-1_0_0}
+ENV BUILD_NUMBER=${BUILD_NUMBER:-1_0_0}
```
Comment (steverendell) [inline Dockerfile:14]: intellij obsolete syntax warning

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/api/PublicApiResource.kt [hunk 2]
```diff
@@ -59,7 +59,7 @@ class PublicApiResource(private val locationService: LocationMappingService) {
 
   @PreAuthorize("hasRole('ROLE_NOMIS_DPS_MAPPING__LOCATIONS__R')")
   @GetMapping("/locations/dps/{dpsLocationId}")
-  @Tag(name = "DPS to NOMIS Mapping lookup")
+  @Tag(name = "NOMIS / DPS Mapping lookup")
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/api/PublicApiResource.kt:62]: tags need to match so these endpoints are shown in the same swagger section

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMapping.kt [hunk 3]
```diff
@@ -10,10 +10,10 @@ import java.util.UUID
 data class CaseNoteMapping(
 
   @Id
-  val dpsCaseNoteId: UUID,
-
   val nomisCaseNoteId: Long,
 
+  val dpsCaseNoteId: UUID,
+
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNoteMapping.kt:16]: We can now have multpile nomis case note ids mapped to the same dps id (when there has been a merge the nomis case notes are copied to the current booking but the DPS notes are not)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt [hunk 4]
```diff
@@ -319,6 +319,36 @@ class CaseNotesMappingResource(private val mappingService: CaseNoteMappingServic
     dpsCaseNoteId: String,
   ): CaseNoteMappingDto = mappingService.getMappingByDpsId(dpsCaseNoteId)
 
+  @GetMapping("/dps-casenote-id/{dpsCaseNoteId}/all")
+  @Operation(
+    summary = "get multiple mappings",
+    description = "Retrieves mappings by DPS id. In case of past merges, there could be > 1 nomis id per dps id. Requires role NOMIS_CASENOTES",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt:325]: I didn't spot any tests for the new endpoint...

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt (hunk 4 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/casenotes/CaseNotesMappingResource.kt:325]: whoah good point ! 🤔

**Pull request merged**